desk/app/channels.hoon

```diff
@@ -723,35 +754,52 @@
   ^+  cor
   ?+    -.sign  !!
       %watch-ack
-    %.  cor
-    ?~  p.sign  same
-    (slog leaf+"Preview failed" u.p.sign)
+    ?~  p.sign  cor
+    %-  (slog leaf+"Preview failed" u.p.sign)
+    ::  treat subscription failures as if we received a %channel-denied
+    ::
+    $(sign [%fact %channel-denied !>(~)])
```

Comment:
Note that this also changes the behavior for old-style reference resolution. But I think this is correct: previously it would just leave the original subscriber hanging without notifying them of the subscription failure in any way.